auto-instrumentation/httpurlconnection/library/src/main/java/io/opentelemetry/instrumentation/library/httpurlconnection/HttpUrlReplacements.java

```diff
@@ -151,21 +151,97 @@ public static String replacementForResponseMessage(URLConnection c) throws IOExc
     }
 
     public static OutputStream replacementForOutputStream(URLConnection c) throws IOException {
-        return replaceThrowable(c, c::getOutputStream, false, false);
+        return replaceThrowable(c, c::getOutputStream, false);
     }
 
     public static InputStream replacementForInputStream(URLConnection c) throws IOException {
-        return replaceThrowable(c, c::getInputStream, false, true);
+        startTracingAtFirstConnection(c);
+
+        InputStream inputStream;
+        try {
+            inputStream = c.getInputStream();
+        } catch (IOException e) {
+            reportWithThrowable(c, e);
+            throw e;
+        }
+
+        if (inputStream == null) {
+            return inputStream;
+        }
+
+        return getInstrumentedInputStream(c, inputStream);
     }
 
     public static InputStream replacementForErrorStream(HttpURLConnection c) {
         startTracingAtFirstConnection(c);
 
         InputStream errorStream = c.getErrorStream();
 
-        reportWithResponseCode(c);
+        if (errorStream == null) {
+            return errorStream;
+        }
+
+        return getInstrumentedInputStream(c, errorStream);
+    }
+
+    private static InputStream getInstrumentedInputStream(
+            URLConnection c, InputStream inputStream) {
+        return new InputStream() {
+            @Override
+            public int read() throws IOException {
+                int res;
+                try {
+                    res = inputStream.read();
+                } catch (IOException e) {
+                    reportWithThrowable(c, e);
+                    throw e;
+                }
+                reportIfDoneOrMarkHarvestable(res);
+                return res;
+            }
+
+            @Override
+            public int read(byte[] b) throws IOException {
+                int res;
+                try {
+                    res = inputStream.read(b);
+                } catch (IOException e) {
+                    reportWithThrowable(c, e);
```

Comment:
Here is where we're reporting any exceptions during input/error stream read. And similar for other read methods covered below.